### PR TITLE
Update problem setup step in oSL solution template to use latest oSL packages to update placeholder values with ui placeholders

### PR DESCRIPTION
**Pull request by ansRMorin**
This PR includes:
- pattern matching callbacks that use the oSL component ids from the ansys-solutions-optislang-frontend-components package to retrieve ui placeholder values
- updates to the problem_setup_step module to include two transaction methods to 1. update the placeholders for data persistence and 2. write the new properties file for analysis

### CODE REVIEW

File: src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/pyproject.toml [hunk 1]
```diff
@@ -48,9 +48,9 @@ ansys-dash-treeview = {version = "0.0.1.dev0", source = "solutions-private-pypi"
 ansys-optislang-core = "^0.3.0"
 ansys-saf-glow = {version = "0.3.0", source = "solutions-private-pypi"}
 ansys-saf-portal = {version = "0.3.0", source = "solutions-private-pypi"}
-ansys-solutions-dash-lib = {version = "^0.4.1", source = "solutions-private-pypi"}
+ansys-solutions-dash-lib = {version = "0.4.3", source = "solutions-private-pypi"}
 ansys-solutions-products-ecosystem = {version = "^0.1.dev0", source = "solutions-private-pypi"}
-ansys-solutions-optislang-frontend-components = {version = "^0.1.dev6", source = "solutions-private-pypi"}
+ansys-solutions-optislang-frontend-components = {version = "0.1.dev7", source = "solutions-private-pypi"}
```
Comment (iazehaf) [inline src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/pyproject.toml:53]: @ansRMorin I suggest we use caret here to relax the version constraint.

File: src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/solution/problem_setup_step.py [hunk 2]
```diff
@@ -48,6 +50,8 @@ class ProblemSetupStep(StepModel):
     }
     step_list: list = []
     node_list: list = []
+    system_hierarchy: list = []
+    step_list: list = []
```
Comment (iazehaf) [inline src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/solution/problem_setup_step.py:54]: @ansRMorin this field already exists at line 49.

File: src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/solution/problem_setup_step.py [hunk 3]
```diff
@@ -48,6 +50,8 @@ class ProblemSetupStep(StepModel):
     }
     step_list: list = []
     node_list: list = []
+    system_hierarchy: list = []
```
Comment (iazehaf) [inline src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/solution/problem_setup_step.py:53]: @ansRMorin what is this for?

**Review by iazehaf — CHANGES_REQUESTED** [review]
@ansRMorin tested against the Oscillator Calibration use case. Received a Dash error when reaching the problem setup page.

Capture

File: src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/solution/problem_setup_step.py (hunk 3 above)
Comment (ansRMorin) [reply src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/solution/problem_setup_step.py:53]: added this by mistake. I removed it!

File: src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/solution/problem_setup_step.py (hunk 2 above)
Comment (ansRMorin) [reply src/ansys/templates/python/osl_solution/{{cookiecutter.__project_name_slug}}/src/ansys/solutions/{{cookiecutter.__solution_name_slug}}/solution/problem_setup_step.py:54]: removed!

**Pull request merged**